app/scripts/controllers/transactions/index.js

```diff
@@ -1219,13 +1219,24 @@ export default class TransactionController extends EventEmitter {
    *  params instead of allowing this method to generate them
    * @param options
    * @param options.estimatedBaseFee
+   * @param actionId
    * @returns {txMeta}
    */
   async createSpeedUpTransaction(
     originalTxId,
     customGasSettings,
     { estimatedBaseFee } = {},
+    actionId,
   ) {
+    // In transaction is found for same action id, do not create a new cancel transaction.
+    if (actionId) {
```

Comment:
We could shave off a couple of lines by handling the null check in `getTransactionWithActionId`. Then we wouldn't need to check for `if (actionId)`

I think we could remove the comment above with this check as well

Reply:
If actionId is not passed, we do not care if function is not idempotent. This is more explicit with above code I think.

Reply:
to be a little clearer here, I was thinking we could update
```
    // In transaction is found for same action id, do not create a new cancel transaction.
    if (actionId) {
      const existingTxMeta =
        this.txStateManager.getTransactionWithActionId(actionId);
      if (existingTxMeta) {
        return existingTxMeta;
      }
    }
```
    
  to 
  
  ```
    // If a transaction is found with the same action id, do not create a new cancel transaction.
    const existingTxMeta = this.txStateManager.getTransactionWithActionId(actionId);
    if (existingTxMeta) {
      return existingTxMeta;
    } 
 ```
 
I still find this proposed solution as explicit and a little more succinct. I'm okay with either though I do prefer less code as long as it's as readable and explicit

Reply:
I would actually prefer first approach

Reply:
k, works with me too